ir/src/symbol_table/symbol_type.rs

```diff
@@ -0,0 +1,36 @@
+use super::{ConstantType, TraceColumns, VariableType};
+use std::fmt::Display;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) enum SymbolType {
+    /// an identifier for a constant, containing its type and value
+    Constant(ConstantType),
+    /// an identifier for a trace column, containing trace column information with its trace
+    /// segment, its size and its offset.
+    TraceColumns(TraceColumns),
+    /// an identifier for a public input, containing the size of the public input array
+    PublicInput(usize),
+    /// an identifier for a periodic column, containing its index out of all periodic columns and
+    /// its cycle length in that order.
+    PeriodicColumn(usize, usize),
+    /// an expression or set of expressions associated with a variable
+    Variable(VariableType),
+    /// an identifier for random value, containing its index in the random values array and its
+    /// length if this value is an array. For non-array random values second parameter is always 1.
+    RandomValuesBinding(usize, usize),
+}
```

Comment:
A comment for future: `SymbolType` and `Value` structs seem very similar. I don't think they can be unified but I wonder if there is a way to tie them together somehow better (e.g., clear way of how `SymbolType` gets converted into `Value`).

Reply:
Good point - it's worth thinking about afterwards. I think there's some more localized refactoring that can still be done, but I focused on trying to clear out the large movements and blocking changes